### DISCUSSION TITLE
How to send binary data?

### DISCUSSION THREAD

**Discussion by q962** [luvit/luvit#1212] [Q&A]
I found in the http.lua:ServerResponse class that all data goes through the self.encode function and I don't know how to send binary data.

There is an example of a static file server in the repository, but it doesn't clearly indicate how to send binary data.

Why is all data encoded using the self.encode function, and no other function is provided to allow binary data to be input?

**Comment by truemedian** [comment]
`self.encode` is responsible for packaging your data into a HTTP message, if you pass binary data to `ServerResponse:write`, then the client will receive that binary data.

**Reply by q962** [reply]
Just use `res:write(blob)`?
